### DISCUSSION TITLE
Power consumption

### DISCUSSION THREAD

**Discussion by achimmm** [raomin/ESPAltherma#409] [General]
Hi community,

I have a question on how to calculate the Overall power consumption.

Am i right to take the value from inverter primary and use this value to calculate Watts on a three Phase Basis? According to the Wiki I would Not use inverter secondary for this calculation.

thanks for and help.

**Comment by MaBeniu** [comment]
Yeaa... I also struggle to get understanding what primary and secondary current values stand for.
Someone who knows more would be appreciated to share their findings.
My unit has heat pump compressor built in water tank and also 2 heating devices (buh1 3kw and buh2 6kw)

**Comment by vincent-bosche** [comment]
You can find it here: https://github.com/raomin/ESPAltherma/wiki/Information-about-Values

The `INV Primary Current (A) `is the current at the Input. It can be single or 3-phase.
The `INV secondary current` is after the power conversion process and a pulsed 3 phase current for the 3 phase motor. It has a variable Voltage (`Voltage (N-phase) (V)`) and Frequency (might be `INV frequency (rps)`).

There is no possibility to get the overall power consumption, since the BUH and controller (incl. motor) has no value to get the power consumption.
Only the compressor itself has the `INV Primary Current (A)`.
Im my case (3HHT) I get a good value when using `INV Primary Current (A) * 230 V * 3 (phases)`. It is a little to hight, but it's quite good!

**Comment by MaBeniu** [comment]
so, I played with the data parameters for some time and I am able to collect nearly correct power consumption. my unit has two heaters 3kw and 6kw power capacity. due to this integration doesn't provide information about heaters I simply put an assumption if heater status is on then I simply add full capacity of the heater into calculation.
![image](https://github.com/user-attachments/assets/16998783-9b29-4927-812e-67b2d86b3c4b)
if someone will find it useful here is template for calculating heating power
```
{% set buh1_state = state_attr('sensor.none_althermasensors', 'BUH Step1') %}
{% set buh2_state = state_attr('sensor.none_althermasensors', 'BUH Step2') %}
{% set primary_current = state_attr('sensor.none_althermasensors', 'INV primary current (A)') | float(0) %}
{% set heating_device_state = state_attr('sensor.none_althermasensors', 'I/U operation mode') %}
{% if buh1_state == 'ON' %}
  {% set buh1_w = 2760 %}
{% else %}
  {% set buh1_w = 0 %}
{% endif %}
{% if buh2_state == 'ON' %}
  {% set buh2_w = 5520 %}
{% else %}
  {% set buh2_w = 0 %}
{% endif %}
{% set current_b = primary_current %}
{% set voltage_b = states('sensor.b_voltage') | float(0) %}
{% set power_factor_b = states('sensor.b_power_factor') | float(1) | abs %}
{% if heating_device_state == 'Heating' %}
  {% set w_b = (voltage_b * current_b * power_factor_b) %}
  {{ w_b + buh1_w + buh2_w }}
{% else %}
  0
{% endif %}
```
copy this template with heating_device_state == 'Heating + DHW' and you will get hot water consumption sensor

The only problem I see so far is a timeshift of approx 3-4 seconds which sometimes in Home Assistant gets treated as 'untracked' consumption.
INV secondary current I found useless at all...

**Comment by fabry1992** [comment]
Im sorry for reviving this thread, but how did you calculate the power factor?

**Reply by MaBeniu** [reply]
:) that impelmentation was 5%-10% inaccurate calculation due to pf was taken from the house mains phase parameters. I ended up installing separate 3phase power meter directly at daikin input and use power draw in watts values from the meter.

**Reply by fabry1992** [reply]
im using just 1 as power factor, but my calculations are like 50% under. Until i find a good deal on a three phase power meter, i at least want to have an estimate, so your 5-10% sounds very good.
